lib/game/client/game-engine.js

```diff
@@ -93,24 +93,27 @@ class GameEngine extends Game
     {
         // @TODO - BETA - Refactor.
         let gameManager = this.uiScene.gameManager;
+        if(GameConst.TYPE_PLAYER !== target.type){
+            return targetName;
+        }
         let showPlayedTime = gameManager.config.get('client/players/playedTime/show');
-        if(0 === showPlayedTime || GameConst.TYPE_PLAYER !== target.type){
+        if (GameConst.SHOW_PLAYER_TIME.NONE === showPlayedTime) {
             return targetName;
         }
         let currentPlayer = gameManager.getCurrentPlayer();
-        let timeText = '';
         let label = gameManager.config.get('client/players/playedTime/label');
-        if(0 < showPlayedTime && currentPlayer.playerId === target.id){
+        let targetDisplayContent = targetName;
+        if(GameConst.SHOW_PLAYER_TIME.ONLY_OWN_PLAYER === showPlayedTime && currentPlayer.playerId === target.id){
             let element = gameManager.gameDom.createElement('p');
             element.innerHTML = label+(currentPlayer.playedTime / 60 / 60).toFixed(1)+'hs';
-            timeText = element.outerHTML;
+            targetDisplayContent += element.outerHTML;
         }
-        if(2 === showPlayedTime && sc.hasOwn(currentPlayer.players, target.id)){
+        if(GameConst.SHOW_PLAYER_TIME.ALL_PLAYERS === showPlayedTime && sc.hasOwn(currentPlayer.players, target.id)){
             let element = gameManager.gameDom.createElement('p');
             element.innerHTML = label+(currentPlayer.players[target.id].playedTime / 60 / 60).toFixed(1)+'hs';
```

Comment:
I'm going to refactor this method, we are doing two process in one method and that is not a good practice, imo. What actions? First showing the targetName, second showing the target's playedTime. I will probably make another method just for adding playedTime into targetDisplay HTML.

Reply:
yeah, we should extract the entire thing outside the game-engine, it's an entire different feature but I was trying to avoid keep adding big refactors on this release, so we can focus on close it and work on all the improvements in the next one